service/tablet_allocator.cc

```diff
@@ -961,49 +961,28 @@ class load_balancer {
                 return tablet_migration_info{kind, gid, src, dst};
             };
 
-            co_await tmap.for_each_sibling_tablets([&] (tablet_desc t1, std::optional<tablet_desc> t2_opt) -> future<> {
-                // Be optimistic about migrating tablets, as if they succeeded.
-                // Merge finalization will have to recheck that all sibling tablets are co-located.
-
-                if (!t2_opt) {
-                    on_internal_error(lblogger, format("Unable to find sibling tablet during co-location, with tablet count {}, for table {}",
-                                                       tmap.tablet_count(), table));
-                }
-                auto t2 = *t2_opt;
+            struct tablet_colocation_info {
+                tablet_desc desc;
+                tablet_replica_set sorted_set;
+            };
+            using check_constraints_f = noncopyable_function<bool(tablet_replica src, tablet_replica dst)>;
```

Comment:
nit: const tablet_replica& to save memory copies as much as possible (in case it isn't optimized by the compiler).

Reply:
This patch just reorganizes the code, maybe do the optimizations in a followup patch.